### PR TITLE
Add wizard

### PR DESCRIPTION
**Pull request by ao-armen**

### CODE REVIEW

File: src/components/HomeScreen/index.module.scss [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #home_screen {
+  height: 800px;
```
Comment (ao-tigran) [inline src/components/HomeScreen/index.module.scss]: What is this?

File: src/App.js [hunk 2]
```diff
@@ -1,59 +1,75 @@
...
-            withCurrencyLabel
-            withSymbol
-            currency=""
-            fractionDigits="3"
-          />
-        )}
-      />
-      {/* ============== END OF auth insensitive routes ================ */}
+const App = () => {
+  const [activeStep, setActiveStep] = useState('step1');
```
Comment (ao-tigran) [inline src/App.js]: Why do we need this one here?

File: src/components/HomeScreen/index.js [hunk 3]
```diff
@@ -27,6 +28,8 @@ const HomeScreen = () => {
         className="custom"
         placeholder="Digits only"
       />
+
+      <WizardContainer />
```
Comment (ao-tigran) [inline src/components/HomeScreen/index.js]: Have a separate route for demonstrating wizard.

File: src/components/WizardContainer/index.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    id: 'step2',
+    content: Step2,
+  },
+];
+
+const WizardContainer = (props) => {
+  const { match: { url }, history } = props;
+  const [activeStep, setActiveStep] = useContext(WizardContext);
+
+  const listenToScroll = () => {};
```
Comment (ao-tigran) [inline src/components/WizardContainer/index.js]: What is this?

File: src/components/WizardContainer/index.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    // clear wizard error
+  }, [activeStep]);
+
+  useEffect(() => {
+    const redirectHome = () => {
+      history.push('/home');
+      // clear wizard error here
+    };
+
+    const pathToStep = () => {
```
Comment (ao-tigran) [inline src/components/WizardContainer/index.js]: Why this method is defined in effect?

File: src/components/WizardContainer/index.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+    const step = pathToStep();
+
+    if (!step) {
+      redirectHome();
+    }
+
+    setActiveStep(activeStep);
+
+    window.addEventListener('scroll', listenToScroll);
```
Comment (ao-tigran) [inline src/components/WizardContainer/index.js]: Why do we add this listener in effect each time? also it seems to be doing nothing?

File: src/components/WizardContainer/index.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+      const { pathname } = history.location;
+      const id = pathname.replace(`${url}/`, '');
+      const [step] = stepContainers.filter((s) => s.id === id);
+
+      return step;
+    };
+
+    const step = pathToStep();
+
+    if (!step) {
```
Comment (ao-tigran) [inline src/components/WizardContainer/index.js]: Why we are doing this here? I think this is a router's responsibility.

File: src/components/WizardContainer/index.js [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+      redirectHome();
+    }
+
+    setActiveStep(activeStep);
```
Comment (ao-tigran) [inline src/components/WizardContainer/index.js]: Shouldn't this be?

`setActiveStep(step);`

File: src/App.js [hunk 9]
```diff
@@ -1,59 +1,75 @@
...
-    </Switch>
-  </GlobalErrProvider>
-);
+        {/* ============== START OF auth routes =========================== */}
+        <AuthenticationRoute
+          path="/"
+          withAuth
+          redirectOnFailure="/login"
+          render={() => (
+            <WizardContext.Provider value={[
```
[the remark this answers is not in the document]
Comment (ao-tigran) [reply src/App.js]: This doesn't look proper to me, what if we have multiple wizards? then what?

File: src/components/WizardContainer/index.js [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+    id: 'step1',
+    content: Step1,
+  },
+  {
+    id: 'step2',
+    content: Step2,
+  },
+];
+
+const WizardContainer = (props) => {
```
Comment (ao-tigran) [inline src/components/WizardContainer/index.js]: Just call this component Wizard, you can avoid conflicts by doing

`import { Wizard as AlbusWizard, Steps, Step } from 'react-albus';`

File: src/components/WizardContainer/index.js [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { Wizard, Steps, Step } from 'react-albus';
+import Step1 from '../Step1';
+import Step2 from '../Step2';
+import WizardContext from '../../context/WizardContext';
+
+import styles from './index.module.scss';
+
+const stepContainers = [
```
Comment (ao-tigran) [inline src/components/WizardContainer/index.js]: Step should be passed to Wizard via property.

File: src/components/WizardContainer/index.js (hunk 5 above)
Comment (ao-armen) [reply src/components/WizardContainer/index.js]: it is a demand of ESLint to place that method in useEffect

**Pull request merged**